src/test/java/com/exasol/errorcodecrawlermavenplugin/ErrorCodeCrawlerMojoIT.java

```diff
@@ -179,8 +180,16 @@ void testDifferentSourcePath() throws IOException, VerificationException {
                 this.projectDir.resolve(alternateSrcPath).toFile());
         final Verifier verifier = getVerifier();
         verifier.executeGoal("error-code-crawler:verify");
-        final String report = Files.readString(this.projectDir.resolve(Path.of("target", "error_code_report.json")));
+        final Path reportPath = this.projectDir.resolve(Path.of("target", "error_code_report.json"));
+        final String report = Files.readString(reportPath);
         assertThat(report, containsString("E-TEST-1"));
+        final ErrorCodeReport parsedReport = new ErrorCodeReportReader().readReport(reportPath);
+        final ErrorMessageDeclaration firstDeclaration = parsedReport.getErrorMessageDeclarations().get(0);
+        assertAll(//
+                () -> assertThat(firstDeclaration.getIdentifier(), equalTo("E-TEST-1")),
+                () -> assertThat(firstDeclaration.getSourceFile(), equalTo("")),
```

Comment:
Would using `null` in this case be an option?

Reply:
So this generator just does not add it. The current behavior of the reader is to read an empty string if the property is missing. We could change that to reading null, but it's in the error-code-model-java repo. So we would need to open a ticket there. It would be a breaking change, so we would also need to update the catalog. So I guess it's not worth the effort...

Reply:
👍